service/src/main/java/zone/cogni/asquare/service/index/DatasetRdfStoreService.java

```diff
@@ -68,4 +69,43 @@ public void executeUpdateQuery(String updateQuery) {
   public void delete() {
     dataset.close();
   }
+
+  /**
+   * <p>
+   * Create and return an in memory dataset copy.
+   * </p>
+   * <p>
+   * The operation runs in a write transaction.
+   * </p>
+   * <p>
+   * It should be noted that this might have a big performance or memory impact when huge datasets are copied.
+   * </p>
+   *
+   * @return a copy of current Dataset
+   */
+  public Dataset copy() {
+    Dataset datasetCopy = DatasetFactory.create();
+
+    try {
+      dataset.getLock().enterCriticalSection(false);
+
+      // copy named models
+      dataset.listNames()
+             .forEachRemaining(name -> {
+               Model namedModel = dataset.getNamedModel(name);
+               datasetCopy.addNamedModel(name, namedModel);
```

Comment:
are we sure this doesn't just use the passed model? Then after the copy a modification would happen in dataset and datasetCopy.